server/src/main/java/org/elasticsearch/ingest/AbstractProcessor.java

```diff
@@ -30,4 +34,36 @@ public String getTag() {
     public String getDescription() {
         return description;
     }
+
+    /**
+     * Helper method to be used by processors that need to catch and log Throwables.
+     * <p>
+     * If trace logging is enabled, then we log the provided message and the full stacktrace
+     * On the other hand if trace logging isn't enabled, then we log the provided message and the message from the Throwable (but not a
+     * stacktrace).
+     * <p>
+     * Regardless of the logging level, we throw an ElasticsearchException that has the context in its message
+     *
+     * @param message A message to be logged and to be included in the message of the returned ElasticsearchException
+     * @param throwable The Throwable that has been caught
+     * @return A new ElasticsearchException whose message includes the passed-in message and the message from the passed-in Throwable. It
+     * will not however wrap the given Throwable.
+     */
+    protected ElasticsearchException logAndBuildException(String message, Throwable throwable) {
+        String cause = throwable.getClass().getName();
+        if (throwable.getMessage() != null) {
+            cause += ": " + throwable.getMessage();
+        }
+        String longMessage = message + ": " + cause;
+        // This method will only be called in exceptional situations, so the cost of looking up the logger won't be bad:
+        Logger logger = LogManager.getLogger(getClass());
+        if (logger.isTraceEnabled()) {
+            logger.trace(message, throwable);
+        } else {
+            logger.warn(longMessage);
```

Comment:
Logging a warning on every ingest processor error seems like it could overwhelm the log, eg if a simple user error is hammered against the system. Why do we need to log user errors?